hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/BlockUtils.java

```diff
@@ -237,6 +238,24 @@ public static void verifyBCSId(KeyValueContainer container, BlockID blockID)
     }
   }
 
+  /**
+   * Verify if request's replicaIndex matches with containerData.
+   *
+   * @param container container object.
+   * @param blockID requested block info
+   * @throws IOException if replicaIndex mismatches.
+   */
+  public static void verifyReplicaIdx(Container container, BlockID blockID)
+      throws IOException {
+    Integer containerReplicaIndex = container.getContainerData().getReplicaIndex();
```

Comment:
Can we verify replicaIndex only when container is EC?
If this check is happening with any container, can you think what happens when containers upgraded from old version where there are no replica index persisted? I think we are not persisting replica index for Ratis containers, so replica index will be null for them? Can you verify auto conversion are safe from Integer to primitive?

Reply:
We could avoid conversion, `null` values, etc. with a dedicated value (e.g. -1) for "not set".

Reply:
Yeah, currently it is assigned replicaIndex to Integer object, so it will attempt to do auto conversion for if check validation in next line. so I am worried whether we are taking care of that, otherwise the code like the below can hit NPE. ( ex:  Integer i = null;  if(i>0) hits NPE)

Reply:
Container uses primitive for replicaIndex, so NPE is not a concern here.

https://github.com/apache/ozone/blob/59560a156d97b0b57ed77772ded3fcaa6fc8e52a/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerData.java#L202-L203

However, it is unnecessarily boxed, only to be able to use `equals` later.

Reply:
container.getContainerData().getReplicaIndex() returns a primitive int. So there is no chance of NPE. I am converting primitive to non primitive. blockID.getReplicaIndex() can return a null value.

Reply:
Got it, thanks for the details.